ext/winevt/winevt_subscribe.c

```diff
@@ -203,8 +203,17 @@ rb_winevt_subscribe_subscribe(int argc, VALUE* argv, VALUE self)
   hSubscription =
     EvtSubscribe(NULL, hSignalEvent, path, query, hBookmark, NULL, NULL, flags);
   if (!hSubscription) {
+    if (hBookmark != NULL) {
+      EvtClose(hBookmark);
+    }
+    if (hSignalEvent != NULL) {
+      CloseHandle(hSignalEvent);
+    }
     status = GetLastError();
     raise_system_error(rb_eWinevtQueryError, status);
+  } else if (winevtSubscribe->subscription != NULL) {
```

Comment:
I prefer omitting the `else` because the previous `if` block never continue the following code.

```
}

if (winevtSubscribe->subscription != NULL) {
```

Reply:
Oh!